src/launch/lib/automation/terragrunt/functions.py

```diff
@@ -1,22 +1,31 @@
-import json
-import logging
 import os
 import subprocess
 from pathlib import Path
 
-from git import Repo
+import click
 
-from launch.config.common import BUILD_TEMP_DIR_PATH
-from launch.constants.launchconfig import LAUNCHCONFIG_NAME
-from launch.lib.automation.environment.functions import install_tool_versions, set_netrc
-from launch.lib.automation.provider.aws.functions import assume_role
-
-logger = logging.getLogger(__name__)
+from launch.cli.j2.render import render
+from launch.config.common import NON_SECRET_J2_TEMPLATE_NAME, SECRET_J2_TEMPLATE_NAME
+from launch.enums.launchconfig import LAUNCHCONFIG_KEYS
 
 
 ## Terragrunt Specific Functions
-def terragrunt_init(run_all=True):
-    logger.info("Running terragrunt init")
+def terragrunt_init(run_all=True, dry_run=True) -> None:
+    """
+    Runs terragrunt init subprocess in the current directory.
```

Comment:
Doesn't need to be addressed as part of this PR, but I'd like to explore whether we have to run TG in the working directory or if there are options to invoke it against other directories.

Ideally we could run the `launch` commands necessary to deploy a given env from the top of the repo and not just in the envs/.../.../... path. We've seen the pattern where we're forced to `cd` around inside a pipeline lead to some unexpected behavior given certain repo structures and tooling (`npm` in my current Azure project so far, but other possibilities exist), so having a command that is location-agnostic (provided you're at least within a valid Launch repo) seems like it would be useful.